desktop/src/main/java/bisq/desktop/main/funds/deposit/DepositView.fxml

```diff
@@ -23,12 +23,22 @@
 <?import javafx.scene.layout.GridPane?>
 <?import javafx.scene.layout.VBox?>
 <?import javafx.geometry.Insets?>
+<?import javafx.scene.layout.HBox?>
+<?import bisq.desktop.components.AutoTooltipLabel?>
+<?import bisq.desktop.components.InputTextField?>
 <VBox fx:id="root" fx:controller="bisq.desktop.main.funds.deposit.DepositView"
       spacing="10" xmlns:fx="http://javafx.com/fxml">
     <padding>
         <Insets bottom="15.0" left="15.0" right="15.0" top="15.0"/>
     </padding>
-
+    <HBox spacing="5.0">
```

Comment:
The usage of FXML is a old left-over from early dev state. We should avoid adding new FXML items and add it in the normal View classes instead. This makes code easier to reuse (e.g. you can make a filter component which is then used every time when you show that filter). Beside better code reusablility and flexibility with more dynamic use cases FXML is slow when getting rendered, thats why we move away from it. We just never spent the time to cleanup the left overs and it would require some more work to remove it completely, so we use it as container still but do all the rest in the view class.

Reply:
I agree - I'm just not sure if this pull request is appropriate place for doing it. `DepositView` and others similar already have FXML items declared. It's probably not good idea to mix both concepts (adding filtering `HBox` in java code while keeping `TableView` and `GridPane` in fxml). To keep this PR compact and monothematic I prefer to keep it as it is and prepare complex refactoring in dedicated pull request after merging this one.

However I'm a bit curious about performance issue with fxml rendering. I thought that thanks of [CachingViewLoader](https://github.com/bisq-network/bisq/blob/7f73df5dfd761a26392fb6ae16adc937ef6bf9da/desktop/src/main/java/bisq/desktop/common/view/CachingViewLoader.java) usage views are parsed once while loading and later it doesn't affect performance. If you know any resources that could help me to better understand this problem I'd be thankful.

Reply:
Ok with me.
Re performance: Yes the CachingViewLoader was introduced to avoid the extra costs each time the view is opened. But there is still the penalty for the first time. If you have a very component heavy UI screen in FXML and then in plain Java code you can see/measure the difference. Parsing FXML does not come for free...

Reply:
As a former frontend dev I always prefer to separate UI from logic - from my perspective it makes code easier to read and maintain. You're right it comes with some costs, don't know how big these costs are - just want to be sure we are not falling into the trap of [premature optimization](https://wiki.c2.com/?PrematureOptimization).

If I'll find time I'll try to measure this problem a bit during refactoring we talked about here.

Reply:
Yes the UI code in Bisq is pretty messy unfortunately. Clear separation can also be done in Java code of course.
It was not premature but a result of experience of laggy UI and then got changed and observed the performance improvement. In Bisq 2 we will not use FXML at all.

Reply:
I've created [dedicated issue](https://github.com/bisq-network/bisq/issues/5989) for refactoring efforts. I'm gonna resolve this conversation.

Reply:
> In Bisq 2 we will not use FXML at all.

I quickly looked into [bisq2 components code](https://github.com/bisq-network/bisq2/tree/main/desktop/src/main/java/bisq/desktop/components) and they looks really promising!